### STORYTELLING CASE STUDY : Airbnb, NYC

#### Problem background
- Suppose that you are working as a data analyst at Airbnb. For the past few months, Airbnb has seen a major decline in revenue. - Now that the restrictions have started lifting and people have started to travel more, Airbnb wants to make sure that it is fully prepared for this change.

#### End Objective
- To prepare for the next best steps that Airbnb needs to take as a business, you have been asked to analyse a dataset consisting of various Airbnb listings in New York.

#### Presentation - I
- Data Analysis Managers: These people manage the data analysts directly for processes and their technical expertise is basic.
- Lead Data Analyst: The lead data analyst looks after the entire team of data and business analysts and is technically sound.

#### Presentation - II
- Head of Acquisitions and Operations, NYC: This head looks after all the property and host acquisitions and operations. Acquisition of the best properties, price negotiation, and negotiating the services the properties offer falls under the purview of this role.
- Head of User Experience, NYC: The head of user experience looks after the customer preferences and also handles the properties listed on the website and the Airbnb app. Basically, the head of user experience tries to optimise the order of property listing in certain neighbourhoods and cities in order to get every property the optimal amount of traction.

In [30]:
# Import the necessary libraries
import warnings
warnings.filterwarnings("ignore")
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [31]:
# Data conversion and Understanding
airbnb = pd.read_csv("AB_NYC_2019.csv")
airbnb.head(5)

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  19-10-2018   
1  Entire home/apt    225               1                 45  21-05-2019   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  05-07-2019   
4  Entire home/apt     80              10                  9  19-11-2018   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [32]:
# Check the rows and columns of the dataset
airbnb.shape

(48895, 16)

- The dataset contains 48895 rows and 16 columns
- Now we have to check whether there are any missing values in the dataset

In [33]:
# Calculating the missing values in the dataset
airbnb.isnull().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

In [6]:
# Now we have the missing values, there are certain columns that are not efficient to the dataset
airbnb.drop(['id','name','last_review'], axis = 1, inplace = True)

In [34]:
# View whether the columns are dropped
airbnb.head(5)

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  19-10-2018   
1  Entire home/apt    225               1                 45  21-05-2019   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  05-07-2019   
4  Entire home/apt     80              10                  9  19-11-2018   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [35]:
airbnb.reviews_per_month.isnull().sum()

10052

In [36]:
# Now reviews per month contains more missing values which should be replaced with 0 respectively
airbnb.fillna({'reviews_per_month':0},inplace=True)

In [37]:
airbnb.reviews_per_month.isnull().sum()

0

In [38]:
# There are no missing values present in reviews_per_month column
# Now to check the unique values of other columns'
airbnb.room_type.unique()

array(['Private room', 'Entire home/apt', 'Shared room'], dtype=object)

In [39]:
len(airbnb.room_type.unique())

3

In [40]:
airbnb.neighbourhood_group.unique()

array(['Brooklyn', 'Manhattan', 'Queens', 'Staten Island', 'Bronx'],
      dtype=object)

In [41]:
len(airbnb.neighbourhood_group.unique())

5

In [42]:
len(airbnb.neighbourhood.unique())

221

In [44]:
airbnb.to_csv(r'C:\Users\Ananta Kumaar V R\Downloads\airbnb_dataset.csv',index=False, header=True)

In [45]:
airbnb.host_id.value_counts().head(10)

219517861    327
107434423    232
30283594     121
137358866    103
16098958      96
12243051      96
61391963      91
22541573      87
200380610     65
7503643       52
Name: host_id, dtype: int64

In [46]:
airbnb2 = airbnb.sort_values(by="calculated_host_listings_count",ascending=False)
airbnb2.head()

id                                            name    host_id  \
39773  30937596      Sonder | 11th Ave | Contemporary 1BR + Gym  219517861   
41463  32240788   Sonder | Stock Exchange | Serene 2BR + Lounge  219517861   
41469  32240966    Sonder | Stock Exchange | Sleek 3BR + Lounge  219517861   
38294  30181945      Sonder | 180 Water | Premier 1BR + Rooftop  219517861   
41468  32240933  Sonder | Stock Exchange | Lively 1BR + Kitchen  219517861   

          host_name neighbourhood_group       neighbourhood  latitude  \
39773  Sonder (NYC)           Manhattan      Hell's Kitchen  40.76037   
41463  Sonder (NYC)           Manhattan  Financial District  40.70782   
41469  Sonder (NYC)           Manhattan  Financial District  40.70620   
38294  Sonder (NYC)           Manhattan  Financial District  40.70771   
41468  Sonder (NYC)           Manhattan  Financial District  40.70726   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
39773  -73.99744  Entire home/apt    185              29                  1   
41463  -74.01227  Entire home/apt    396               2                  8   
41469  -74.01192  Entire home/apt    498               2                  8   
38294  -74.00641  Entire home/apt    229              29                  1   
41468  -74.01060  Entire home/apt    229               2                  2   

      last_review  reviews_per_month  calculated_host_listings_count  \
39773  10-06-2019               1.00                             327   
41463  24-06-2019               2.12                             327   
41469  11-06-2019               2.50                             327   
38294  29-05-2019               0.73                             327   
41468  23-06-2019               0.77                             327   

       availability_365  
39773               332  
41463               289  
41469               255  
38294               219  
41468               351